token/program-2022/src/state.rs

```diff
@@ -80,6 +103,55 @@ impl Pack for Mint {
     }
 }
 
+/// MintWithTransferFee data.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq)]
+pub struct MintWithTransferFee {
```

Comment:
I think we'll need a `withheld_amount: u64` in the mint as well, so that there's somewhere for an `AccountWithTransferFee`'s `withheld_amount` to go when the `Burn` instruction is used.

Notably, a `Burn` of 0 could be used to empty the `withheld_amount` field of an account.

Reply:
This is why I initially had `TransferWithheld`, to make it clear that you can transfer out your own withheld tokens to the fee owner.  I'm not a huge fan of the Burn behavior since it's unclear.

On the other hand, if we store a withheld amount in the mint, then there's no reason to ever withhold amounts in the token accounts.  It's only useful if we want to support unchecked `Transfer` for `Account2`, which isn't possible.  That would resolve pretty much all of the withholding issues that have come up.  What do you think?

Reply:
> I'm not a huge fan of the additional Burn behavior and would prefer it done with a separate instruction.

Yeah, a new `Close` variant is probably cleaner than `Burn(0)`